parsers/glade.c

```diff
@@ -0,0 +1,110 @@
+/*
+*
+*   Copyright (c) 2015, Masatake YAMATO
+*   Copyright (c) 2015, Red Hat, K.K.
+*
+*   This source code is released for free distribution under the terms of the
+*   GNU General Public License version 2 or (at your option) any later version.
+*
+*   This module contains functions for generating tags for diff files (based on Sh parser).
+*/
+
+#include "general.h"	/* must always come first */
+#include "debug.h"
+#include "options.h"
+#include "parse.h"
+#include "read.h"
+#include "routines.h"
+
+
+typedef enum {
+	R_CLASS_WIDGET,
+} gladeClassRole;
+
+typedef enum {
+	R_HANDLER_HANDLER,
+} gladeHandleRoler;
+
+static roleDesc GladeClassRoles [] = {
+	{ TRUE, "widget", "specifed as a widget constructor" },
+};
+
+static roleDesc GladeHandlerRoles [] = {
+	{ TRUE, "handler", "specifed as a callback for signal emission" },
+};
+
+typedef enum {
+	K_ID, K_CLASS, K_HANDLER,
+} gladeKind;
+
+static kindOption GladeKinds [] = {
+	{ TRUE,  'i', "id",	  "identifiers" },
+
+	/* These two are appeared on names in C source code. */
+	{ TRUE,  'c', "class",	  "classes",
+	  .referenceOnly = TRUE, ATTACH_ROLES (GladeClassRoles) },
+	{ TRUE,  'h', "handler",  "handlers",
+	  .referenceOnly = TRUE, ATTACH_ROLES (GladeHandlerRoles) },
+};
+
+static tagXpathTable gladeXpathTable[] = {
+	{ "///glade-interface//widget//@id",
+	  K_ID,    ROLE_INDEX_DEFINITION },
+	{ "///glade-interface//widget//@class",
+	  K_CLASS, R_CLASS_WIDGET },
+	{ "///glade-interface//signal//@handler",
+	  K_HANDLER, R_HANDLER_HANDLER },
+};
+
+static void
+findGladeTags (void)
+{
+	xmlDocPtr doc;
+	xmlNode *root;
+	xmlXPathContextPtr ctx;
+
+	findRegexTags ();
```

Comment:
Drive the regex parser here.
See Units/simple-glade.d/args.ctags in these commits.